src-tauri/src/mod_management/mod.rs

```diff
@@ -370,24 +373,40 @@ pub async fn fc_download_mod_and_install(
     );
 
     // Download the mod
-    let f = match thermite::core::manage::download_file(download_url, path.clone()) {
-        Ok(f) => f,
-        Err(e) => return Err(e.to_string()),
-    };
+    let f = thermite::core::manage::download_file(download_url, path.clone())
+        .map_err(|err| err.to_string())?;
```

Comment:
This might sound a bit like bike shedding but I highly suggest to make additional PRs with small changes like these that technically are not part of the main PR.

Cause then I can merge them early and reduce the size of the diff of this PR which in turn makes reviewing this PR easier ^^